src/sentry/buffer/redis.py

```diff
@@ -155,6 +181,24 @@ def incr(self, model, columns, filters, extra=None):
             - Perform a set (last write wins) on extra
         - Add hashmap key to pending flushes
         """
+
+        if _local_buffers is not None:
+            with _local_buffers_lock:
+                if _local_buffers is not None:
+                    frozen_filters = tuple(sorted(filters.items()))
+                    key = (frozen_filters, model)
+
+                    stored_columns, stored_extra = _local_buffers.get(key, ({}, None))
+
+                    for k, v in columns.items():
+                        stored_columns[k] = stored_columns.get(k, 0) + v
+
+                    if extra is not None:
+                        stored_extra = extra
```

Comment:
Just to confirm, we basically ignore the previous value of `stored_extra` when reassigning here, right?

Reply:
Yes. It's unclear which behavior is expected by the caller of `buffer.incr` but this should be fine for the invocations in outcomes consumer.